### ISSUE TITLE
Rename ContentRichEntity to Resource

### ISSUE THREAD

**Issue by niklasnatter** [sulu/SuluContentBundle#181]
I would like to propose to rename the `ContentRichEntityInterface` to `ResourceInterface`. 

I see the following reasons for doing this:
- Sulu uses the term `Resource` for an entity that is displayed in the administration interface
- We already have methods that use the `Resource` term: `DimensionContentInterface::getResourceKey`, `RoutableInterface::getResourceId` 
- `Resource` is less bulky than `ContentRichEntity` and would make the bundle code easier to read and easier to talk about
- Together with https://github.com/sulu/SuluContentBundle/issues/90, this simplifies the core terms/entities/concepts of the bundle and therefore makes it easier to understand for new developers

**Comment by alexander-schranz** [comment]
I personally think we should get ride of the `ContentRichEntityInterface` /  `ResourceInterface` completely. As it should be possible to add Content to every Doctrine Entity, also if it comes from a third party bundle where we are not able to add an Interface our self. 

Currently the `ContentRIchEntityInterface` has the following methods:

 - `getId()`
 - `getDimensionContents()`
 - `createDimensionContent()`
 - `addDimensionContent(DimensionContentInterface $dimensionContent)`
 - `removeDimensionContent(DimensionContentInterface $dimensionContent):`
 
And the `DimensionContentInterface` reference to it with the following methods:

 - `getResource()`
 - `getResourceKey()`
 - `getResourceId` (RoutableInterface)
 
 So we could replace the exist function:
 
  - `getId()` -> `getResourceId()`
 - `getDimensionContents()` -> DimensionContentRepository should be used here
 - `createDimensionContent()` -> DimensionContentInterface::create function given the resource object `create(object $resource)`
 - `addDimensionContent(DimensionContentInterface $dimensionContent)` -> not needed if a relation on doctrine exist can be called in `create` could call the add on the resource
 - `removeDimensionContent(DimensionContentInterface $dimensionContent):` -> never used or called by us

**Comment by niklasnatter** [comment]
Yes, getting rid of the `ContentRichEntityInterface` would be even better! But I think its also more work unfortunately 🙂  
Do you want to create a separate issue for that?

**Comment by alexander-schranz** [comment]
Created a seperate issue for it: #183
